modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -1032,14 +1053,29 @@ public void updateLease(long leaseStartTime) {
             AbstractWriteBatch writeBatch = PartitionDataHelper.requireWriteBatch();
 
             try {
-                byte[] leaseBytes = new byte[Long.BYTES];
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                outputStream.write(longToBytes(leaseStartTime));
+
+                byte[] primaryReplicaNodeIdBytes = stringToBytes(primaryReplicaNodeId);
+                assert primaryReplicaNodeIdBytes.length < Byte.MAX_VALUE :
+                        format("Primary replica node id bytes length exceeds the limit [length={}].",
+                                primaryReplicaNodeIdBytes.length);
+                outputStream.write((byte) primaryReplicaNodeIdBytes.length);
+                outputStream.write(primaryReplicaNodeIdBytes);
 
-                putLongToBytes(leaseStartTime, leaseBytes, 0);
+                byte[] primaryReplicaNodeNameBytes = stringToBytes(primaryReplicaNodeName);
+                assert primaryReplicaNodeNameBytes.length < Byte.MAX_VALUE;
```

Comment:
looks like newly introduced limitation, should it be documented?
Now I can successfully create a cluster of nodes having 127 characters in their names

Reply:
Well, seems that 127 is too small. Reworked to use int instead.